### ISSUE TITLE
docs: Missing Component testing section?

### ISSUE THREAD

**Issue by MentalGear** [vitest-dev/vitest#2125]
### Describe the bug

The main site as well as the docs state under **Features**
"Component testing for Vue, React, Svelte, Lit and more"

But neither on the page, nor when searching for "Component" is a section that actually describes how component testing with, for example svelte, is done.

### Reproduction

https://vitest.dev/guide/features.html#smooth-integration-with-ui-frameworks

Search for "component"

### System Info

```shell
/
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by bjarkihall** [comment]
The docs might refer to https://github.com/vitest-dev/vitest/tree/main/examples

**Comment by MentalGear** [comment]
Yes, a section in the docs how to integrate `@testing-library/svelte` would be much appreciated!

**Comment by bjarkihall** [comment]
Yes, one idea would be to describe (in general) what the examples are showcasing/testing in this section: https://vitest.dev/guide/#examples , e.g. just include the key word "Component testing" and "Testing-Library" for example, so it's easier to map the caption on the front page to these examples and also to increase SEO in this regard.

"Svelte Component Testing" should hopefully give you results linking to this section and clicking the "features caption on the front-page" you mentioned in the ticket should probably be a clickable URL that points to that section.

These should be relatively easy to add to the docs if you would like to try it out and the maintainers agree with the suggestions.

**Comment by sheremet-va** [comment]
> These should be relatively easy to add to the docs if you would like to try it out and the maintainers agree with the suggestions.

I think we can add a new section in Guide, describing best practices with component testing, and give a link to examples.

**Comment by qrhfz** [comment]
there should be an integration test for vue

**Comment by akoskm** [comment]
I see that in https://github.com/vitest-tests/browser-examples/commit/27640045d6e8747c2ed01dce9449cf42ab211e16 the official react example was updated to use `vitest-browser-react` instead of `@testing-library/react`. @sheremet-va I think it would be super helpful to have an example with RTL on your examples page, given its popularity. If you're accepting PRs, I'm happy to create an example for this.

**Comment by sheremet-va** [comment]
> I see that in [vitest-tests/browser-examples@2764004](https://github.com/vitest-tests/browser-examples/commit/27640045d6e8747c2ed01dce9449cf42ab211e16) the official react example was updated to use `vitest-browser-react` instead of `@testing-library/react`. [<img alt="" width="16" height="16" src="https://avatars.githubusercontent.com/u/16173870?u=a93924a94b9840bc6b76fac48b50f80177df979b&amp;v=4&amp;size=80">@sheremet-va](https://github.com/sheremet-va?rgh-link-date=2025-02-12T06%3A10%3A54.000Z) I think it would be super helpful to have an example with RTL on your examples page, given its popularity. If you're accepting PRs, I'm happy to create an example for this.

The RTL example was intentionally removed to not cause confusion to readers. Vitest provides a built-in API that works differently than RTL, but has a similar naming convention. For reasoning see https://github.com/vitest-dev/vitest/discussions/5828#discussioncomment-10314822

**Comment by akoskm** [comment]
Thanks, @sheremet-va. I see that `vitest-browser-react` is your recommended way to test React applications. I thought the example would be valuable since most apps probably still use RTL (we also decided to go with RTL after trying browser mode, which we will re-evaluate later). But I understand you don't need the example for RTL. Thanks!

**Comment by blmille1** [comment]
Is all that we need a link (or individual links to the various options) from [FeaturesList.vue](https://github.com/vitest-dev/vitest/blob/6d64a3f6e64b4e4e7c3b50f6a3a5b932903951e7/docs/.vitepress/components/FeaturesList.vue#L10) to [the vitest-tests browser examples](https://github.com/vitest-tests/browser-examples/tree/main/examples)?
